lib/src/segy.c

```diff
@@ -562,26 +562,12 @@ static int segy_seek( segy_file* fp,
 #endif //HAVE_MMAP
 
     int err;
-#if LONG_MAX == LLONG_MAX
-    assert( pos <= LONG_MAX );
-    err = fseek( fp->fp, (long)pos, SEEK_SET );
+#ifdef HAVE_FSEEKO
+    err = fseeko( fp->fp, pos, SEEK_SET );
```

Comment:
Do we need to ensure to always use a 64-bit integer or does it not matter? 

From the Microsoft documentation, I understand that `_fseeki64` will always use a 64-bit integer to represent the offset.

The Linux (POSIX) [definition](https://pubs.opengroup.org/onlinepubs/9699919799/) seems to be vague what `off_t` is. From some [discussions on Stackoverflow](https://stackoverflow.com/a/1035750) it appears that one might have to enforce 64-bit offset types by a compile definition. I assume we are happy with the POSIX definition of `fseeko` because it would just use and cast to the correct type itself (?). Is this understanding correct?